### PR TITLE
DIT-610 Save Email Template with event

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/uk/gov/hmrc/tariffclassificationfrontend/connector/EmailConnector.scala [hunk 1]
```diff
@@ -35,4 +36,9 @@ class EmailConnector @Inject()(configuration: AppConfig, client: HttpClient) {
     client.POST(url = url, body = e).map(_ => Unit)
   }
 
+  def generate[T](e: Email[T])(implicit hc: HeaderCarrier, writes: Format[T]): Future[EmailTemplate] = {
+    val url = s"${configuration.emailRendererUrl}/templates/${e.templateId}"
+    client.POST[Map[String, T], EmailTemplate](url, Map("parameters" -> e.parameters))
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/connector/EmailConnector.scala]: I didn't think there was any point making a model for this

**Comment by ed0906** [comment]
Hold off merging this for a min, I just realised the email body is Base64 encoded

File: app/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService.scala [hunk 2]
```diff
@@ -62,10 +62,25 @@ class CasesService @Inject()(appConfig: AppConfig, auditService: AuditService, e
...
     for {
-      updated <- connector.updateCase(caseUpdating)
-      _ <- emailService.sendCaseCompleteEmail(updated)
-        .recover({case t: Throwable => Logger.error(s"Could not send Complete Case Email for case [${updated.reference}]", t)})
-      _ <- addEvent(original, updated, operator, Some("The applicant was sent an Email Confirmation with their reference"))
+      // Update the case
+      updated: Case <- connector.updateCase(caseUpdating)
+
+      // Send the email
+      message <- emailService.sendCaseCompleteEmail(updated)
```
Comment (gokyo) [inline app/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService.scala]: This code is unreadable.

Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService.scala]: Generally its more constructive to make a suggestion on how to improve ;)

**Pull request merged**